client/search_box.js

```diff
@@ -0,0 +1,71 @@
+TemplateController('search_box', {
+  private: {
+    currentSearchFieldIndex: 0 // email by default
+  },
+  onCreated(){
+    this.formatOK = function (fieldName, format,value) {
+      let hasError = true;
+      let errMssg = "";
+      switch (format) {
+        case "string":
+          if (value.length === 0 || //accept empty
+              isNaN(value) )
+          {
+            hasError = false;
+          }
+          break;
+        case "integer":
+          if (!isNaN(value))
+          {
+            hasError = false;
+          }
+          break;
+        default:
+          errMssg = "incorrect format";
+
+      }
+      if (hasError)
+      {
+        errMssg = "not a valid "+fieldName;
+      }
+      return errMssg;
+    },
+
+    this.goSearch =  function(event) {
```

Comment:
you can add the `formatOk` and `goSearch` functions to the `private` object to make them available as helpers on the template instance! No need to define them within `onCreated`